specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json

```diff
@@ -519,17 +519,55 @@
           "type": "string"
         },
         "status": {
-          "description": "The status of the email",
+          "description": "The status of the email. Any value other than Delivered is considered failed.",
           "type": "string",
           "enum": [
-            "Failed",
+            "Bounced",
             "Delivered",
+            "Failed",
             "FilteredSpam",
-            "Quarantined"
+            "Quarantined",
+            "Suppressed"
           ],
           "x-ms-enum": {
             "name": "AcsEmailDeliveryReportStatus",
-            "modelAsString": true
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Bounced",
+                "description": "Hard bounce detected while sending the email"
+              },
+              {
+                "value": "Delivered",
+                "description": "The email was delivered"
+              },
+              {
+                "value": "Failed",
+                "description": "The email failed to be delivered"
+              },
+              {
+                "value": "FilteredSpam",
+                "description": "The message was identified spam and was rejected or blocked (not quarantined)."
+              },
+              {
+                "value": "Quarantined",
+                "description": "The message was quarantined (as spam, bulk mail, or phishing). For more information, see Quarantined email messages in EOP (EXCHANGE ONLINE PROTECTION)."
+              },
+              {
+                "value": "Suppressed",
+                "description": "The email was suppressed"
+              }
+            ]
+          }
+        },
+        "deliveryStatusDetails": {
```

Comment:
Actually, now that this is an object, you'll need to define it as a separate definition. I'd suggest calling it AcsEmailDeliveryStatusDetails.

Reply:
I see other objects being defined in place. Any particular reason why I need to define it separately?

Reply:
There shouldn't be other objects like this other than ones where additionalProperties is set to true. This will not work with the code gen as it doesn't know what to name the type.

Reply:
Actually, nevermind. Just saw the definition

Reply:
Updating

Reply:
SDK needs to generate a class, and anonymous models forces the SDK generator to come up with a name by itself if it wants to generate. For this reason, this pattern is now forbideen, and codegen shall fail until you give an actual name.

Reply:
Can we actually call this AcsEmailDeliveryReportStatusDetails instead? Sorry for not suggesting this initially!

Reply:
Sure. Updated.